project-example/src/main.rs

```diff
@@ -15,10 +15,15 @@ use std::collections::BTreeMap;
 use structopt::StructOpt;
 use tap::*;
 
-mod generated_code;
-mod lib;
+mod connection_handler;
+mod opt;
 
-use generated_code::example;
+// HACK: Include `generated.rs` directly rather than treating it like a regular
+// module file because it internally declares a module named `generated`, which
+// causes clippy to complain about nested modules with the same name. We'll
+// likely end up using `include!` if we switch to doing code generation in a
+// build script, but the generated code will be at a more reasonable place.
+include!("generated.rs");
```

Comment:
The large downside of this is that you no longer get IDE features about generated code from the IntelliJ Rust plugin (maybe RLS too?). 

I'd rather add explicit clippy exclusions than this

Reply:
I'm surprised IntelliJ Rust doesn't support this, the Rust compiler is able to correctly give compiler warnings/errors pointing into `generated.rs`, though I'm not sure how well RLS handles it (RLS is pretty flaky about completion in general, so I don't really notice if things aren't working correctly). I'll switch to a regular `mod` statement and add a clippy inclusion instead 👍

This issue might need to be revisited as part of #56, though. I'm pretty sure the `include!` macro is the only way to pull in code generated in a `build.rs` script, or at least that's the way recommended in the Rust docs.

Reply:
I was able to confirm that RLS *does* still work when using `include!`, which is nifty.